overlord/fdestate/fdemgr.go

```diff
@@ -599,6 +599,22 @@ func (m *FDEManager) CheckRecoveryKey(rkey keys.RecoveryKey, containerRoles []st
 	return nil
 }
 
+// systemEncrypted reports whether FDE is enabled on the system.
+// It returns true if FDE is enabled, or false otherwise.
+func (m *FDEManager) systemEncrypted() (bool, error) {
+	var s FdeState
+	err := m.state.Get(fdeStateKey, &s)
+	if err != nil {
+		return false, err
+	}
+
+	// Systems with FDE enabled will have a primary key with the exception
+	// of a limited legacy cases. See comment on type FdeState field
+	// PrimaryKeys for details.
+	_, hasPrimaryKey := s.PrimaryKeys[0]
```

Comment:
Shall we be more defensible?
```suggestion
    if s.PrimaryKeys == nil {
        return false, nil
    }
	_, hasPrimaryKey := s.PrimaryKeys[0]
```

Reply:
`PrimaryKeys` is a map. Am I missing something?

Reply:
Not related to previous comment, but I wonder if one day we change this 0 ID for the main primary key. Maybe it is better to check if there are *any* primary key.

Reply:
I missed that this is a map.